lib/error.sh

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Description: Error handling functions.
+
+# Write specified message to stderr.
+stderr()
+{
+	local msg="$*"
+
+	echo >&2 "$msg"
+}
+
+# Simplified version of die that should be called by functions that
+# could fail as part of the normal "die()" code path. Required to
+# avoid infinite recursion.
+_fatal()
+{
+	echo >&2 "FATAL: $msg"
+	exit 1
+}
+
+# Canonicalize the specified path (which must be valid).
+resolve_path()
+{
+	local file="${1:-}"
+	[ -z "$file" ] && _fatal "need file to resolve"
+
+	local path
+
+	if grep -q "^/" <<< "$file"
+	then
+		path=$(readlink -e "$file")
+	else
+		path=$(readlink -e "$PWD/$file")
+	fi
+
+	echo "$path"
+}
+
+show_proc_hierarchy()
+{
+	local pid="${$}"
+
+	local details
+
+	stderr "process-hierarchy:"
+
+	local -i i=0
+
+	while true
+	do
+		local details
+		local msg
+
+		local current=''
+
+		[ "${pid}" = "${$}" ] && current=", current='yes'"
+
+		details=$(ps --no-headers -p "$pid" -o ppid,cmd)
+
+		local ppid=$(echo "$details" | awk '{print $1}')
```

Comment:
Could you elaborate on what you are trying to do here? This looks like you are trying to strip whitespace?